File: src/web/magmac/app/lang/web/TypescriptFunctionStatement.ts
import { MapNode } from "../../../../magmac/app/compile/node/MapNode";
import { Node } from "../../../../magmac/app/compile/node/Node";
import { FunctionStatement } from "../../../../magmac/app/lang/common/FunctionStatement";
export class TypescriptFunctionStatement {
	TypescriptFunctionStatement(child : TypescriptFunctionSegmentValue) : public {0( 0);;}
	serialize() : Node {return new MapNode( "statement").withNodeSerialized( "child", 0.child);;}
}

File: src/java/magmac/app/lang/web/TypescriptFunctionStatement.java
package magmac.app.lang.web;

import magmac.app.compile.node.MapNode;
import magmac.app.compile.node.Node;
import magmac.app.lang.common.FunctionStatement;

public final class TypescriptFunctionStatement extends FunctionStatement<TypescriptFunctionSegmentValue> implements TypescriptLang.TypescriptFunctionSegment {
    public TypescriptFunctionStatement(TypescriptFunctionSegmentValue child) {
        super(child);
    }

    @Override
    public Node serialize() {
        return new MapNode("statement").withNodeSerialized("child", this.child);
    }
}
